src/rgw/rgw_rest_s3.h

```diff
@@ -928,14 +928,13 @@ class RGWSelectObj_ObjStore_S3 : public RGWGetObj_ObjStore_S3
   virtual int get_params() override;
 
 private:
-  void encode_short(char &buff, short s, int &i);
+  void encode_short(char *buff, short s, int &i);
 
-  void encode_int(char &buff, u_int32_t s, int &i);
+  void encode_int(char *buff, u_int32_t s, int &i);
 
   int create_header_records(char *buff);
 
-  // the parameters are according to CRC-32 algorithm and its aligned with AWS-cli checksum
```

Comment:
why use dynamic allocation?

Reply:
previously , the buffer was allocated on stack(1000 bytes)
issue been discussed (not to use stack for this sizes), thus , its now dynamic.